### PR TITLE
Re-include UTR length calculations and plot

### PR DESCRIPTION
**Pull request by ninsch3000**

### CODE REVIEW

**Comment by AngryMaciek** [comment]
> _No description provided._

Please provide descriptions of PRs; bullet points with changes are OK

File: PAQR2/PAQR/README.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+- singular_pas_expression.tsv
+- tandem_pas_expression.tsv
+- tandem_pas_relative_usage.tsv
+- relative_pas_positions.tsv
+- tandem_pas_expression_normalized.tsv
+- filtered_pas_expression.tsv
+- filtered_pas_positions.tsv
+- filtered_pas_usage.tsv
+- weighted_avg_exon_lengths.tsv
+- weigthed_avg_exon_lengths.pdf
```
Comment (AngryMaciek) [inline PAQR2/PAQR/README.md:16]: I am not sure if it makes sense to list a intermediate files<?>
If you think this is a good idea then OK.

File: PAQR2/PAQR/scripts/infer-pas-expression.py [hunk 2]
```diff
@@ -2055,6 +2055,20 @@ def main(options):
...
+                "end",
+                "pas",
+                "score",
+                "strand",
+                "polyAsite_exon_idx",
+                "nr_polyAsites_on_exon",
+                "exon",
+                "gene",
+                "\t".join(options.sample_names) ) )
+
```
Comment (AngryMaciek) [inline PAQR2/PAQR/scripts/infer-pas-expression.py:2071]: I assume that the header is the same as for _expressions_ file, correct? That is good then.

File: PAQR2/PAQR/Snakefile [hunk 3]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
         --filtered-positions {output.TSV_filtered_pas_positions} \
+        --filtered-usage {output.TSV_filtered_usage} \
         1> {log.LOG_local_stdout} 2> {log.LOG_local_stderr}
         """
+
+#-------------------------------------------------------------------------------
+# get weighted average exon length
+#-------------------------------------------------------------------------------
+rule weighted_average_exon_length:
+    input:
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:527]: rule header does not match other rules, please adjust the format

File: PAQR2/PAQR/Snakefile [hunk 4]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+        1> {output.TSV_exon_lengths} \
+        2> {log.LOG_local_stderr}
+        """
+
+#-------------------------------------------------------------------------------
+# plot the cumulative distribution for the weighted average exon lengths
+# per sample
+#-------------------------------------------------------------------------------
+rule plot_average_exon_length:
+    ##LOCAL##
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:591]: header does not match the rest, please adjust format

File: PAQR2/PAQR/Snakefile [hunk 5]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+
+    singularity:
+        "docker://zavolab/rzavolab:4.0.0"
+    shell:
+        '''
+        Rscript {input.SCRIPT_} \
+        --pdf={output.PDF_exon_lengths} \
+        --file={input.TSV_exon_lengths} \
+        1> {log.LOG_local_stdout} 2> {log.LOG_local_stderr}
+        '''
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:643]: GitHub complains

File: PAQR2/PAQR/Snakefile [hunk 6]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+            "{PAQ_output_dir}",
+            "cluster_log",
+            "PAQ_exon_lenghts.log"
+        )
+
+    log:
+        LOG_local_stdout = os.path.join(
+            "{PAQ_output_dir}",
+            "local_log",
+            "PAQ_exon_lengths.stdout.log"
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:556]: `PAQ` + rule name

File: PAQR2/PAQR/Snakefile [hunk 7]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+        ),
+        LOG_local_stderr = os.path.join(
+            "{PAQ_output_dir}",
+            "local_log",
+            "PAQ_exon_lengths.stderr.log"
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:561]: `PAQ` + rule name

File: PAQR2/PAQR/Snakefile [hunk 8]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+        )
+
+    benchmark:
+        os.path.join(
+            "{PAQ_output_dir}",
+            "local_log",
+            "PAQ_exon_lengths.benchmark.log"
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:568]: `PAQ` + rule name . benchmark

File: PAQR2/PAQR/Snakefile [hunk 9]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+    output:
+        TSV_exon_lengths = os.path.join(
+            "{PAQ_output_dir}",
+            "weighted_avg_exon_lengths.tsv"
+        )
+    params:
+        LOG_cluster_log = os.path.join(
+            "{PAQ_output_dir}",
+            "cluster_log",
+            "PAQ_exon_lenghts.log"
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:549]: `PAQ` + rule name

File: PAQR2/PAQR/Snakefile [hunk 10]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+    output:
+        PDF_exon_lengths = os.path.join(
+            "{PAQ_output_dir}",
+            "weighted_avg_exon_lengths.pdf"
+        )
+    params:
+        LOG_cluster_log = os.path.join(
+            "{PAQ_output_dir}",
+            "cluster_log",
+            "PAQ_plot_exon_lenghts.log"
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:610]: `PAQ` + rule name

File: PAQR2/PAQR/Snakefile [hunk 11]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+        )
+
+    log:
+        LOG_local_stdout = os.path.join(
+            "{PAQ_output_dir}",
+            "local_log",
+            "PAQ_plot_exon_lengths.stdout.log"
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:617]: `PAQ` + rule name

File: PAQR2/PAQR/Snakefile [hunk 12]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+        ),
+        LOG_local_stderr = os.path.join(
+            "{PAQ_output_dir}",
+            "local_log",
+            "PAQ_plot_exon_lengths.stderr.log"
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:622]: `PAQ` + rule name

File: PAQR2/PAQR/Snakefile [hunk 13]
```diff
@@ -498,7 +513,131 @@ rule PAQ_filter_on_expression:
...
+        )
+
+    benchmark:
+        os.path.join(
+            "{PAQ_output_dir}",
+            "local_log",
+            "PAQ_plot_exon_lengths.benchmark.log"
```
Comment (AngryMaciek) [inline PAQR2/PAQR/Snakefile:629]: `PAQ` + rule name .  benchmark

File: PAQR2/PAQR/env/r.yml [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+name: r-4-0-0
+channels:
+ - conda-forge
+dependencies:
+ - r-base=4.0
+ - r-ggplot2
+ - r-gplots
+ - r-optparse
+ - r-rcolorbrewer
+...
```
Comment (AngryMaciek) [inline PAQR2/PAQR/env/r.yml:11]: GitHub complains

File: PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py [hunk 15]
```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
```
Comment (AngryMaciek) [inline PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py:1]: If you are running a script in _conda_ then specifying a python interpreter out of the conda env does not make sense!
Remove the shebang line.

File: PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+"""
+My template:
+"""
```
Comment (AngryMaciek) [inline PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py:4]: ?

File: PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+
+__date__ = "2016-07-07"
+__author__ = "Ralf Schmidt"
+__email__ = "ralf.schmidt@unibas.ch"
+__license__ = "GPL"
```
Comment (AngryMaciek) [inline PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py:9]: which version?

File: PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+                    default=False,
+                    help="Be loud!")
+
+parser.add_argument("--relativePos",
+                    dest="relativePos",
+                    help="path and name of the file with the positions of the poly(A) sites relative to the exon")
+
+parser.add_argument("--relUsage",
+                    dest="rel_usages",
+                    help="path and name of the file with the relative usages")
```
Comment (AngryMaciek) [inline PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py:30]: This script takes the filtered relative usages, right?
I assume that as you added the header to the unfiletered relative usages file but the header is propagated here too? Then I guess you modified this original script to account for the headers too?

File: PAQR2/PAQR/scripts/filter-tables.py [hunk 19]
```diff
@@ -73,15 +85,17 @@ def main():
 
     try:
         positions = pd.read_csv(options.pas_positions, sep="\t", index_col=0, header=None)
+        usage = pd.read_csv(options.pas_usage, sep="\t")
         expression = pd.read_csv(options.normalized_expression, sep="\t")
-        samples = expression.columns.values[10:]
```
Comment (AngryMaciek) [inline PAQR2/PAQR/scripts/filter-tables.py:77]: we dont need this variable anymore?

File: PAQR2/PAQR/scripts/plot-ecdfs.R [hunk 20]
```diff
@@ -0,0 +1,171 @@
+suppressPackageStartupMessages(library("optparse"))
```
Comment (AngryMaciek) [inline PAQR2/PAQR/scripts/plot-ecdfs.R:1]: Missing info lines about the author, license, email etc.

**Review by AngryMaciek — CHANGES_REQUESTED** [review]
Generally OK; minor corrections required, mostly formatting

File: PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py (hunk 18 above)
Comment (ninsch3000) [reply PAQR2/PAQR/scripts/calculate-average-3pUTR-length.py:30]: yes, see line 55

File: PAQR2/PAQR/scripts/filter-tables.py (hunk 19 above)
Comment (ninsch3000) [reply PAQR2/PAQR/scripts/filter-tables.py:77]: Never needed it

File: PAQR2/PAQR/scripts/plot-ecdfs.R (hunk 20 above)
Comment (ninsch3000) [reply PAQR2/PAQR/scripts/plot-ecdfs.R:1]: Good point, but I don't know those

Comment (AngryMaciek) [reply PAQR2/PAQR/scripts/plot-ecdfs.R:1]: I think Ralf is the autor, we can set MIT license and Mihaela can be the contact person w/ email.

**Pull request closed without merging**